### PR TITLE
rusk-recovery: handle multiple keyset

### LINKED ISSUE #727: Use different keys for harness genesis contract
**Describe "Why" this is needed**
To allow `dusk-blockchain` to run test-harness with dummy keys

### PR DESCRIPTION
**Pull request by herr-seppia**
#### Added
- Allow to build the genesis state using prebuilt contracts (with `-c` flag)

#### Changed
- `-t` flag uses a different provisioners key set while building the state

#### Chores:
- Always print the current network state root
- Replace `lazy_static` with `once_cell`

Resolves #727

### CODE REVIEW

File: rusk-recovery/src/bin/state.rs [hunk 1]
```diff
@@ -46,6 +46,10 @@ struct Cli {
     /// Sets different levels of verbosity
     #[clap(short, long, parse(from_occurrences))]
     verbose: usize,
+
+    /// Use prebuilt contracts when building the state from scratch .
```
Comment (ureeves) [inline rusk-recovery/src/bin/state.rs:50]:
```suggestion
    /// Use prebuilt contracts when building the state from scratch.
```

File: rusk-recovery/src/keys.rs [hunk 2]
```diff
@@ -123,7 +120,7 @@ pub fn run_circuit_keys_checks(
 }
 
 pub fn exec(keep_keys: bool) -> Result<(), Box<dyn std::error::Error>> {
-    lazy_static::initialize(&PUB_PARAMS);
+    Lazy::force(&PUB_PARAMS);
```
Comment (ureeves) [inline rusk-recovery/src/keys.rs:123]: Is this really necessary?

**Review by ureeves — APPROVED** [review]
LGTM. Two nits

File: rusk-recovery/src/keys.rs (hunk 2 above)
Comment (herr-seppia) [reply rusk-recovery/src/keys.rs]: Weird...
Just noticed that without this the crs is never checked (and therefore never cached)

**Pull request merged**